### DISCUSSION TITLE
c++ problem solving pls

### DISCUSSION THREAD

**Discussion by asd123123123123123123** [community/community#35668] [Other Feature Feedback, Questions, & Ideas]
Given the following function definition:

int sum_down(int x)
{
	if (x >= 0)
	{
		x = x - 1;
		int y = x + sum_down(x);
		return y + sum_down(x);
	}
	else
	{
		return 1;
	}
}

1.Rewrite the function, so that it is free of recursion. I.e. give an iterative definition on the foundation of a loop-construct.TIP: First transform the recursive definition, so that you have just a single recursive call.

2.Is it OK to switch the type of the parameter x to double?Discuss your decision / give an argumentation.

3.Is it OK to switch the type of the parameter x to unsigned int?Discuss your decision / give an argumentation.

4.Is it OK to switch the function head to int sum_down(const int x)?Discuss your decision / give an argumentation.

i'm a beginner c++.. pls solve this problem...

**Comment by queenofcorgis** [comment]
Hi there @asd123123123123123123 ,
The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your post was flagged and will be removed because it violated our [Community Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md), we ask that members contribute in a positive and constructive way. Asking for help on homework or test questions without context will likely discourage engagement and responses.

If this content is posted again, your Community account will be subject to an immediate and indefinite suspension.
For additional guidance, please review our [Community Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service).

Thank you!
